diff --git a/build/core/pathmap.mk b/build/core/pathmap.mk
index fd0f5a4..a1ef4ea 100644
--- a/build/core/pathmap.mk
+++ b/build/core/pathmap.mk
@@ -95,6 +95,7 @@ FRAMEWORKS_BASE_SUBDIRS := \
 	    telephony \
 	    wifi \
 	    keystore \
+	    fmradio \
 	 )
 
 #
diff --git a/build/target/product/core_base.mk b/build/target/product/core_base.mk
index 14c350a..e693cf7 100644
--- a/build/target/product/core_base.mk
+++ b/build/target/product/core_base.mk
@@ -32,6 +32,7 @@ PRODUCT_PACKAGES += \
     libbcc \
     libfilterpack_imageproc \
     libgabi++ \
+    libanalogradiobroadcasting \
     libkeystore \
     libmdnssd \
     libnfc_ndef \
