diff --git a/hardware/libhardware_legacyaudio/AudioPolicyManagerBase.cpp b/hardware/libhardware_legacyaudio/AudioPolicyManagerBase.cpp
index 722bbd3..4ea127e 100644 (file)
--- a/hardware/libhardware_legacy/audio/AudioPolicyManagerBase.cpp
+++ b/hardware/libhardware_legacy/audio/AudioPolicyManagerBase.cpp
@@ -2563,6 +2563,9 @@ AudioPolicyManagerBase::device_category AudioPolicyManagerBase::getDeviceCategor
         case AUDIO_DEVICE_OUT_BLUETOOTH_SCO_HEADSET:
         case AUDIO_DEVICE_OUT_BLUETOOTH_A2DP:
         case AUDIO_DEVICE_OUT_BLUETOOTH_A2DP_HEADPHONES:
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+        case AUDIO_DEVICE_OUT_FM:
+#endif
             return DEVICE_CATEGORY_HEADSET;
         case AUDIO_DEVICE_OUT_SPEAKER:
         case AUDIO_DEVICE_OUT_BLUETOOTH_SCO_CARKIT:
@@ -3405,7 +3408,7 @@ const struct StringToEnum sDeviceNameToEnumTable[] = {
     STRING_TO_ENUM(AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET),
     STRING_TO_ENUM(AUDIO_DEVICE_OUT_USB_DEVICE),
     STRING_TO_ENUM(AUDIO_DEVICE_OUT_USB_ACCESSORY),
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
     STRING_TO_ENUM(AUDIO_DEVICE_OUT_FM),
     STRING_TO_ENUM(AUDIO_DEVICE_OUT_FM_TX),
 #endif
@@ -3421,7 +3424,7 @@ const struct StringToEnum sDeviceNameToEnumTable[] = {
     STRING_TO_ENUM(AUDIO_DEVICE_IN_ANC_HEADSET),
 #endif
     STRING_TO_ENUM(AUDIO_DEVICE_IN_AUX_DIGITAL),
-#ifdef QCOM_HARDWARE
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
     STRING_TO_ENUM(AUDIO_DEVICE_IN_FM_RX),
     STRING_TO_ENUM(AUDIO_DEVICE_IN_FM_RX_A2DP),
 #endif
diff --git a/hardware/libhardware_legacyinclude/hardware_legacy/AudioSystemLegacy.h b/hardware/libhardware_legacyinclude/hardware_legacy/AudioSystemLegacy.h
index 9832052..971a633 100644 (file)
--- a/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
+++ b/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
@@ -61,11 +61,13 @@ enum audio_source {
     AUDIO_SOURCE_CAMCORDER = 5,
     AUDIO_SOURCE_VOICE_RECOGNITION = 6,
     AUDIO_SOURCE_VOICE_COMMUNICATION = 7,
-#if defined (STE_FM)
-    AUDIO_SOURCE_FM_RX = 8,
-    AUDIO_SOURCE_MAX = AUDIO_SOURCE_FM_RX,
+    AUDIO_SOURCE_REMOTE_SUBMIX = 8,
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+    AUDIO_SOURCE_FM_RX = 9,
+    AUDIO_SOURCE_FM_RX_A2DP = 10,
+    AUDIO_SOURCE_MAX = AUDIO_SOURCE_FM_RX_A2DP,
 #else
-    AUDIO_SOURCE_MAX = AUDIO_SOURCE_VOICE_COMMUNICATION,
+    AUDIO_SOURCE_MAX = AUDIO_SOURCE_REMOTE_SUBMIX,
 #endif
     AUDIO_SOURCE_LIST_END  // must be last - used to validate audio source type
 };
@@ -250,9 +252,17 @@ public:
         DEVICE_OUT_ANLG_DOCK_HEADSET = 0x800,
         DEVICE_OUT_DGTL_DOCK_HEADSET = 0x1000,
 #ifdef QCOM_HARDWARE
-        DEVICE_OUT_ANC_HEADSET = 0x2000,
-        DEVICE_OUT_ANC_HEADPHONE = 0x4000,
-        DEVICE_OUT_PROXY = 0x8000,
+        DEVICE_OUT_USB_ACCESSORY = 0x2000,
+        DEVICE_OUT_USB_DEVICE = 0x4000,
+#endif
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+        DEVICE_OUT_FM = 0x8000,
+        DEVICE_OUT_FM_TX = 0x10000,
+#endif
+#ifdef QCOM_HARDWARE
+        DEVICE_OUT_ANC_HEADSET = 0x20000,
+        DEVICE_OUT_ANC_HEADPHONE = 0x40000,
+        DEVICE_OUT_PROXY = 0x80000,
         DEVICE_OUT_DEFAULT = DEVICE_OUT_SPEAKER,
 #else
         DEVICE_OUT_DEFAULT = 0x8000,
@@ -263,14 +273,31 @@ public:
                 DEVICE_OUT_BLUETOOTH_A2DP_SPEAKER | DEVICE_OUT_AUX_DIGITAL |
                 DEVICE_OUT_ANLG_DOCK_HEADSET | DEVICE_OUT_DGTL_DOCK_HEADSET |
 #ifdef QCOM_HARDWARE
+                DEVICE_OUT_USB_ACCESSORY | DEVICE_OUT_USB_DEVICE |
                 DEVICE_OUT_ANC_HEADSET | DEVICE_OUT_ANC_HEADPHONE |
                 DEVICE_OUT_PROXY |
 #endif
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+                DEVICE_OUT_FM | DEVICE_OUT_FM_TX |
+#endif
                 DEVICE_OUT_DEFAULT),
         DEVICE_OUT_ALL_A2DP = (DEVICE_OUT_BLUETOOTH_A2DP | DEVICE_OUT_BLUETOOTH_A2DP_HEADPHONES |
                 DEVICE_OUT_BLUETOOTH_A2DP_SPEAKER),
+#ifdef QCOM_HARDWARE
+        DEVICE_OUT_ALL_USB = (DEVICE_OUT_USB_ACCESSORY | DEVICE_OUT_USB_DEVICE),
 
         // input devices
+        DEVICE_IN_COMMUNICATION = 0x100000,
+        DEVICE_IN_AMBIENT = 0x200000,
+        DEVICE_IN_BUILTIN_MIC = 0x400000,
+        DEVICE_IN_BLUETOOTH_SCO_HEADSET = 0x800000,
+        DEVICE_IN_WIRED_HEADSET = 0x1000000,
+        DEVICE_IN_AUX_DIGITAL = 0x2000000,
+        DEVICE_IN_VOICE_CALL = 0x4000000,
+        DEVICE_IN_BACK_MIC = 0x8000000,
+        DEVICE_IN_ANC_HEADSET = 0x10000000,
+#else
+        // input devices
         DEVICE_IN_COMMUNICATION = 0x10000,
         DEVICE_IN_AMBIENT = 0x20000,
         DEVICE_IN_BUILTIN_MIC = 0x40000,
@@ -279,15 +306,17 @@ public:
         DEVICE_IN_AUX_DIGITAL = 0x200000,
         DEVICE_IN_VOICE_CALL = 0x400000,
         DEVICE_IN_BACK_MIC = 0x800000,
-#ifdef QCOM_HARDWARE
-        DEVICE_IN_ANC_HEADSET = 0x10000000,
-        DEVICE_IN_PROXY = 0x20000000,
-        DEVICE_IN_ANLG_DOCK_HEADSET = 0x40000000,
+        DEVICE_IN_DEFAULT = 0x80000000,
 #endif
-#if defined (STE_FM)
-        DEVICE_IN_FM_RX = 0x1000000,
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+        DEVICE_IN_FM_RX = 0x20000000,
+        DEVICE_IN_FM_RX_A2DP = 0x40000000,
+#endif
+#ifdef QCOM_HARDWARE
+        DEVICE_IN_DEFAULT = DEVICE_IN_BUILTIN_MIC,
+        DEVICE_IN_ANLG_DOCK_HEADSET = 0x80000000,
+        DEVICE_IN_PROXY = DEVICE_IN_ANLG_DOCK_HEADSET,
 #endif
-        DEVICE_IN_DEFAULT = 0x80000000,
 
         DEVICE_IN_ALL = (DEVICE_IN_COMMUNICATION | DEVICE_IN_AMBIENT | DEVICE_IN_BUILTIN_MIC |
                 DEVICE_IN_BLUETOOTH_SCO_HEADSET | DEVICE_IN_WIRED_HEADSET | DEVICE_IN_AUX_DIGITAL |
@@ -295,8 +324,8 @@ public:
 #ifdef QCOM_HARDWARE
                 DEVICE_IN_ANC_HEADSET | DEVICE_IN_PROXY | DEVICE_IN_ANLG_DOCK_HEADSET |
 #endif
-#if defined (STE_FM)
-                DEVICE_IN_FM_RX |
+#if defined(QCOM_FM_ENABLED) || defined(STE_FM)
+                DEVICE_IN_FM_RX | DEVICE_IN_FM_RX_A2DP |
 #endif
                 DEVICE_IN_DEFAULT)
     };
-- 
1.8.5.3