diff --git a/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h b/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
index a1e9c64..9832052 100644
--- a/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
+++ b/hardware/libhardware_legacy/include/hardware_legacy/AudioSystemLegacy.h
@@ -74,6 +74,13 @@ enum audio_source {
     AUDIO_SOURCE_MAX = AUDIO_SOURCE_VOICE_COMMUNICATION,
 #endif
 
+#if defined (STE_FM)
+    AUDIO_SOURCE_FM_RX = 8,
+    AUDIO_SOURCE_MAX = AUDIO_SOURCE_FM_RX,
+#else
+     AUDIO_SOURCE_MAX = AUDIO_SOURCE_VOICE_COMMUNICATION,
+#endif
+
     AUDIO_SOURCE_LIST_END  // must be last - used to validate audio source type
 };
 
@@ -316,6 +316,10 @@ public:
         DEVICE_IN_DEFAULT = 0x80000000,
 #endif
 
+#if defined (STE_FM)
+        DEVICE_IN_FM_RX = 0x1000000,
+#endif
+
         DEVICE_IN_ALL = (DEVICE_IN_COMMUNICATION | DEVICE_IN_AMBIENT | DEVICE_IN_BUILTIN_MIC |
                 DEVICE_IN_BLUETOOTH_SCO_HEADSET | DEVICE_IN_WIRED_HEADSET | DEVICE_IN_AUX_DIGITAL |
 #ifdef QCOM_HARDWARE
@@ -323,6 +327,9 @@ public:
                 DEVICE_IN_FM_RX | DEVICE_IN_FM_RX_A2DP | DEVICE_IN_DEFAULT |
                 DEVICE_IN_ANLG_DOCK_HEADSET | DEVICE_IN_PROXY)
 #else
+#if defined (STE_FM)
+                DEVICE_IN_FM_RX |
+#endif
                 DEVICE_IN_VOICE_CALL | DEVICE_IN_BACK_MIC | DEVICE_IN_DEFAULT)
 #endif
     };
-- 
1.8.5.3

