diff --git a/external/bluetooth/bluez/android/hal-bluetooth.c b/external/bluetooth/bluez/android/hal-bluetooth.c
index a01229a..d6a5afe 100644
--- a/external/bluetooth/bluez/android/hal-bluetooth.c
+++ b/external/bluetooth/bluez/android/hal-bluetooth.c
@@ -232,7 +232,7 @@ static void handle_pin_request(void *buf, uint16_t len)
 	DBG("");
 
 	if (bt_hal_cbacks->pin_request_cb)
-		bt_hal_cbacks->pin_request_cb(addr, name, ev->class_of_dev);
+		bt_hal_cbacks->pin_request_cb(addr, name, ev->class_of_dev, 0);
 }
 
 static void handle_ssp_request(void *buf, uint16_t len)
diff --git a/external/bluetooth/bluez/android/client/if-bt.c b/external/bluetooth/bluez/android/client/if-bt.c
index 8dcffea..02488a1 100644
--- a/external/bluetooth/bluez/android/client/if-bt.c
+++ b/external/bluetooth/bluez/android/client/if-bt.c
@@ -229,7 +229,7 @@ static void pin_request_answer(char *reply)
 }
 
 static void pin_request_cb(bt_bdaddr_t *remote_bd_addr, bt_bdname_t *bd_name,
-								uint32_t cod)
+						uint32_t cod, uint8_t secure)
 {
 	/* Store for command completion */
 	bt_bdaddr_t2str(remote_bd_addr, last_remote_addr);
